// Copyright 2016 The Fuchsia Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package cobalt.analyzer;

option go_package = "analyzer_service";

import "google/protobuf/empty.proto";
import "observation_batch.proto";

// The Analyzer receives observations from the shufflers.  Reports based on
// these observations are periodically generated and published.
service Analyzer {
  // Shuffler -> Analyzer RPC
  rpc AddObservations (ObservationBatch) returns (google.protobuf.Empty) {}
}
